### PR TITLE
feat: help with fees questions

### PR DESCRIPTION
**Pull request by SebFlippence**
### JIRA link ###

Part of:

* [`NFDIV-445`](https://tools.hmcts.net/jira/browse/NFDIV-445) English Version - Ask if help is needed to pay the application fee
* [`NFDIV-247`](https://tools.hmcts.net/jira/browse/NFDIV-247) Welsh Version - Ask if help is needed to pay the application fee
* [`NFDIV-449`](https://tools.hmcts.net/jira/browse/NFDIV-449) English Version - Have you already applied for help with fees for your divorce / to end your civil partnership
* [`NFDIV-450`](https://tools.hmcts.net/jira/browse/NFDIV-450) English Version - Display the Help with Fees application page

### Change description ###

* Adds 3 pages with part translation
* Handles form subfields
* Fixes issue with substep navigation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/steps/sequence/help-paying-have-ref-no/content.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+              type: 'text',
+              attributes: {
+                maxLength: 11,
+              },
+              classes: 'govuk-!-width-one-third',
+              label: l => l.enterRefNo,
+              hint: l => `
+                <p class="govuk-label">${l.refReceivedWhenApplied}</p>
+                ${l.refExample}`,
+              validator: (value: string | Record<string, string>): void | string => {
```
Comment (velimd) [inline src/main/steps/sequence/help-paying-have-ref-no/content.ts:61]: maybe have this as a method in validations file?

File: src/main/steps/sequence/help-paying-have-ref-no/content.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                    return 'invalidUsedExample';
+                  }
+                }
+              },
+            },
+          },
+        },
+        { label: l => l.no, value: 'No' },
+      ],
+      validator: isFieldFilledIn,
```
Comment (velimd) [inline src/main/steps/sequence/help-paying-have-ref-no/content.ts:82]: never seen this before where function has argument, how does it work? is there a name for this approach?

File: src/main/steps/sequence.ts [hunk 3]
```diff
@@ -76,4 +79,28 @@ export const sequence: Step[] = [
...
+        field: 'alreadyAppliedForHelpPaying',
+        when: res => res.helpPayingNeeded === 'Yes',
+        url: HELP_PAYING_HAVE_YOU_APPLIED,
+        subSteps: [
+          {
+            id: 'help-paying-need-to-apply',
+            title: 'You need to apply for help with your divorce fees',
+            field: 'helpWithFeesRefNo',
+            when: res => res.alreadyAppliedForHelpPaying === 'No',
+            url: HELP_PAYING_NEED_TO_APPLY,
```
Comment (velimd) [inline src/main/steps/sequence.ts:100]: final page?

File: src/main/steps/sequence/help-paying-have-ref-no/content.ts (hunk 1 above)
Comment (SebFlippence) [reply src/main/steps/sequence/help-paying-have-ref-no/content.ts]: 👍 Fixed in 05558f2.

File: src/main/steps/sequence.ts (hunk 3 above)
Comment (SebFlippence) [reply src/main/steps/sequence.ts:100]: 👍 Fixed in 05558f2.

File: src/main/steps/sequence/help-paying-have-ref-no/content.ts (hunk 2 above)
Comment (SebFlippence) [reply src/main/steps/sequence/help-paying-have-ref-no/content.ts:67]: This assigns the function to the variable so essentially they're both the same:

`validator: isFieldFilledIn` === `validator: (value) => isFieldFilledIn(value)`

**Review by velimd — APPROVED** [review]

**Pull request merged**